sqlmesh/core/model/kind.py

```diff
@@ -422,14 +424,19 @@ class FullKind(_ModelKind):
 class _SCDType2Kind(_ModelKind):
     dialect: str = ""
     unique_key: SQLGlotListOfFields
-    valid_from_name: SQLGlotString = "valid_from"
-    valid_to_name: SQLGlotString = "valid_to"
+    valid_from_name: SQLGlotColumn = Field(exp.column("valid_from"), validate_default=True)
+    valid_to_name: SQLGlotColumn = Field(exp.column("valid_to"), validate_default=True)
     invalidate_hard_deletes: SQLGlotBool = False
     time_data_type: exp.DataType = Field(exp.DataType.build("TIMESTAMP"), validate_default=True)
 
     forward_only: SQLGlotBool = True
     disable_restatement: SQLGlotBool = True
 
+    # Remove once Pydantic 1 is deprecated
+    _always_validate_column = field_validator(
```

Comment:
Why didn't we need this for `SQLGlotString`?

Reply:
We need this for when we get a default value. The validator for `SQLGlotString` just ensures that we get a string result which we know will happen since the default was a string. In this case we need to make sure that the default is normalized to the dialect so we always need to validate.